File: C#-part-2/FrequentNumber/FrequentNumber.cs
/*
Problem 9. Frequent number

Write a program that finds the most frequent number in an array.
 */
using System;


	class FrequentNumber
	{
		public static void Main(string[] args)
		{
			Console.WriteLine("Write lenght of an array:");
			int n = int.Parse(Console.ReadLine());
			int[] arr = new int[n];
			
			for (int i = 0; i < n; i++)
            {
            Console.Write("Enter element[{0}]: ", i);
            arr[i] = int.Parse(Console.ReadLine());
            }
			
			Console.WriteLine(string.Join(",", arr));
            Console.WriteLine();
		    
			int biggestCounter = 0;
            int frequentNum = 0;			
			int counter = 1;
			for (int i = 0; i < arr.Length - 1; i++)
            {
            if (arr[i] == arr[i + 1])
            {
                counter++;
            }
            else 
            {
                if (counter > biggestCounter)
                {
                    biggestCounter = counter;
                    frequentNum = arr[i];
                }
                counter = 1;
            }
            }
            Console.WriteLine("{0} ({1} times)", frequentNum, biggestCounter);
			
		}
	}

File: Homeworks/2. Web Design and UI technlogies/3. JavaScript-Part-1/05.Arrays/06. Most frequent number/FrequentNumber.js
/**Problem 6. Most frequent number
 Write a script that finds the most frequent number in an array.
 */
function MostFrq(args) {
    var input = +args[0].split('\n'),
        n = input[0],
        numbers = input.slice(1),
        current = numbers[0],
        counter,
        bestNum = 0,
        bestCounter = 0,
        result = [];

    numbers.sort();

    for (counter = 0; counter < n; counter += 1) {
        if (numbers[counter] == numbers[counter + 1]) {
            current += 1;
        }
        else {
            if (current > bestCounter) {
                bestCounter = current;
                bestNum = numbers[counter];
                current = 1;
            }
            else {
                current = 1;
            }
        }
    }

    if (current > bestCounter) {
        bestCounter = counter;
        bestNum = numbers[n];
    }

    result[0] = bestNum;
    result[1] = bestCounter;

    return console.log(bestNum + " (" + bestCounter + " times)");
}